File: Assignments.API/Assignments.API/Models/ToDoItem.cs
using System;

namespace Assignments.API.Models
{
    public class ToDoItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool IsArchived { get; set; }
        public bool IsRepeating { get; set; }
        public int TaskTypeId { get; set; }
        public TaskType TaskType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime DueDate { get; set; }
        public DateTime? CompleteDate { get; set; }
    }
}

File: Assignments.Client/src/app/todoitem.ts
export type TaskType = {
  id: number;
  name: string;
};
export interface TodoItem {
  id: number;
  name: string;
  description: string;
  isArchived: boolean;
  isRepeating: boolean;
  taskType: TaskType;
  startDate: string;
  dueDate: string;
  completeDate: string;
}
